src/test/java/com/waffle/dredes/level/LevelGen.java

```diff
@@ -241,10 +320,39 @@ public void tryAdd(String center, Room.Direction direction, String neighbor) {
         }
 
         if(c != null && n != null) {
-            c.addNeighbor(direction, n);
+            c.addNeighbor(direction, n, true);
+        }
+    }
+
+    public void tryAdd(String center, Room.Direction direction, String neighbor, boolean flipped) {
+        Room c = roomLoader.getRoom(center);
+        Room n = roomLoader.getRoom(neighbor);
+
+        if(c == null) {
+            Constants.LOGGER.logSevere("Room '" + center + "' was not found!");
```

Comment:
You should always log from the class you are sending the message from: easier to trace without stacktrace

Reply:
Great to know! I'll definitely take this into account to improve my logger.

Reply:
Good way to remember is to keep your loggers `private static final`